lib/truffle/socket/unix_server.rb

```diff
@@ -27,7 +27,7 @@
 class UNIXServer < UNIXSocket
   def initialize(path)
     @no_reverse_lookup = self.class.do_not_reverse_lookup
-    @path              = path
+    @path              = Truffle::Type.check_null_safe(path)
```

Comment:
FWIW this and the MRI fix break support for Linux abstract sockets (`man 7 unix`).
But it seems a weird and rarely used feature, so let's follow what MRI does.